containers/ftp/get-ecs-task-public-ip

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+TMPFILE=ecstaskpubliciptempfile
+AWS=/usr/local/bin/aws
+
+function err() {
+    echo "E: $*" >>/dev/stderr
+}
+
+function try() {
+    eval $*
+    if [ $? -ne 0 ]; then
+        err "Error while evaluating \"$*\", exiting..."
+        exit 1
+    fi
+}
+
+try rm -f $TMPFILE
+
+try curl -s ${ECS_CONTAINER_METADATA_URI_V4}/task -o $TMPFILE
+
+ECS_CLUSTER=$(cat $TMPFILE | jq -r '.Cluster'|cut -d/ -f2)
+ECS_TASK=$(cat $TMPFILE | jq -r '.TaskARN'|cut -d/ -f2)
+
+if [[ -z "${ECS_CLUSTER}" ]]  || [[ -z "${ECS_TASK}" ]] ; then
+    err "Missing Task or Cluster ID, exiting..."
+    exit 1
+fi
+
+try $AWS ecs describe-tasks --tasks $ECS_TASK --cluster $ECS_CLUSTER  > $TMPFILE
+ENI_ID=$(cat $TMPFILE | jq -r '.tasks[].attachments[].details[] | select(.name =="networkInterfaceId") | .value ')
+
+
+if [[ -z "${ENI_ID}" ]] ; then
+    err "Missing ENI ID, exiting..."
+    exit 1
+fi
+
+try $AWS ec2 describe-network-interfaces  --filters Name=network-interface-id,Values=$ENI_ID > $TMPFILE
+
+ECS_TASK_PUBLIC_IP=$(cat $TMPFILE | jq -r '.NetworkInterfaces[0].Association.PublicIp')
+
+if [[ -z "${ECS_TASK_PUBLIC_IP}" ]] ; then
+    err "Missing Public IP, exiting..."
+    exit 1
+fi
+
+echo $ECS_TASK_PUBLIC_IP
```

Comment:
This was easier to get working, and having that IP is really beside the point of what the ftp script does - it's info about the containing ENV and so makes more sense to me for the application logic to receive this as an ENV var.

Reply:
Oh yeah, I didn't mean to incorporate this into the FTP script, just to write this script in Ruby rather than bash, to keep everything in the same language. Very roughly:

```ruby
require 'net/http'
require 'json'

ecs = Aws::ECS::Client.new
ec2 = Aws::EC2::Client.new

meta = JSON.parse(Net::HTTP.get("#{ENV['ENV']}/task", '/index.html'))
exit 1 if !meta['Cluster'] || !meta['TaskARN']

ecs_res = ecs.describe_tasks({ cluster: meta['Cluster'], tasks: [meta['TaskARN']] })
details = ecs_res['tasks'][0]['attachments'][0]['details']
eni_id = details.find { |d| d['name'] == 'networkInterfaceId'}['value']
exit 1 if !eni_id

ec2_res = ec2.describe_network_interfaces({ network_interface_ids: [eni_id] })
ip = ec2_res['NetworkInterfaces'][0]['Association']['PublicIp']
exit 1 if !ip

print ip
```

We've already got Ruby and the AWS SDK available, so I'd make an argument to leverage them and end up with code that's easier to read and maintain. I also think using the Ruby gem that we're explicitly versioning is somewhat safer than the CLI coming from the Dockerfile that could drift over time as the image gets rebuilt